src/Microsoft.DotNet.Build.Tasks/UpdateProjectDependencies.cs

```diff
@@ -0,0 +1,126 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NuGet.Versioning;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class UpdateProjectDependencies : Task
+    {
+        [Required]
+        public ITaskItem ProjectJson { get; set; }
+
+        public string OldPrerelease { get; set; }
+
+        public string NewPrerelease { get; set; }
+
+        public string PackageId { get; set; }
+
+        public string OldVersion { get; set; }
+
+        public string NewVersion { get; set; }
+
+        public override bool Execute()
+        {
+            if (new[] { OldPrerelease, NewPrerelease }.All(s => !string.IsNullOrEmpty(s)))
```

Comment:
Lets do some basic parameter validation.  If NewPrerelease is set and OldPrerelease is not, log an error.
```
if (!String.IsNullOrWhiteSpace(NewPrerelease))
{
  if (!String.IsNullOrWhiteSpace(OldPrerelease))
  {
    ReplacePrerelease();
  }
  else
  {
    // log error
  }
}
```

Do something similar for version mode.

If neither are specified, error.  If both are specified, error, since it isn't clear if PackageId should apply to the pre-release.